lib/Transforms/Scalar/LoopDeletion.cpp

```diff
@@ -170,9 +170,11 @@ bool LoopDeletion::runOnLoop(Loop *L, LPPassManager &LPM) {
   // Don't remove loops for which we can't solve the trip count.
   // They could be infinite, in which case we'd be changing program behavior.
   ScalarEvolution &SE = getAnalysis<ScalarEvolution>();
-  const SCEV *S = SE.getMaxBackedgeTakenCount(L);
-  if (isa<SCEVCouldNotCompute>(S))
-    return Changed;
+  // HLSL Change begin - remove loops even cannot solve the trip count.
+  //const SCEV *S = SE.getMaxBackedgeTakenCount(L);
+  ////if (isa<SCEVCouldNotCompute>(S))
+  //  return Changed;
+  // HLSL Change end.
```

Comment:
It's more complicated than I implied here. This will only change the behavior where infinite loops with no side effects come into play. that might trip up tests intended to check behavior on TDRs, but apart from that, I can't think of anyone who would complain about having a GPU hang automatically removed.